markdown_it/ruler.py

```diff
@@ -25,6 +25,19 @@ def __init__(self, src: str, md, env):
         self.env = env
         self.md = md
 
+    @property
+    def src(self):
+        return self._src
+
+    @src.setter
+    def src(self, value):
+        self._src = value
+        self._srcCharCode = [ord(c) for c in self.src]
+
+    @property
+    def srcCharCode(self):
+        # To avoid assigning srcCharCode
+        return self._srcCharCode
```

Comment:
```suggestion
```

I think we have to remove this property and keep `srcCharCode` writeable. We write it directly in `StateBlock.__init__`. It seems to me like a performance optimization, but @chrisjsewell knows better.